backend/src/v2/component/launcher_v2_test.go

```diff
@@ -0,0 +1,127 @@
+		name                string
+		executorInput       *pipelinespec.ExecutorInput
+		wantErr             bool
+		expectedOutputParam float64
+	}{
+		{
+			"happy pass",
+			&pipelinespec.ExecutorInput{
+				Inputs: &pipelinespec.ExecutorInput_Inputs{
+					ParameterValues: map[string]*structpb.Value{"a": structpb.NewNumberValue(1), "b": structpb.NewNumberValue(2)},
+				},
+				Outputs: &pipelinespec.ExecutorInput_Outputs{
+					Parameters: map[string]*pipelinespec.ExecutorInput_OutputParameter{
+						"Output": {OutputFile: "/tmp/kfp/outputs/Output"},
+					},
+					OutputFile: "/tmp/kfp_outputs/output_metadata.json",
+				},
+			},
+			false,
+			3,
+		},
+		{
+			"use default value",
+			&pipelinespec.ExecutorInput{
+				Inputs: &pipelinespec.ExecutorInput_Inputs{
+					ParameterValues: map[string]*structpb.Value{"b": structpb.NewNumberValue(2)},
+				},
+				Outputs: &pipelinespec.ExecutorInput_Outputs{
+					Parameters: map[string]*pipelinespec.ExecutorInput_OutputParameter{
+						"Output": {OutputFile: "/tmp/kfp/outputs/Output"},
+					},
+					OutputFile: "/tmp/kfp_outputs/output_metadata.json",
+				},
+			},
+			false,
+			7,
+		},
+		{
+			"missing parameter",
+			&pipelinespec.ExecutorInput{
+				Inputs: &pipelinespec.ExecutorInput_Inputs{
+					ParameterValues: map[string]*structpb.Value{},
+				},
+				Outputs: &pipelinespec.ExecutorInput_Outputs{
+					Parameters: map[string]*pipelinespec.ExecutorInput_OutputParameter{
+						"Output": {OutputFile: "/tmp/kfp/outputs/Output"},
+					},
+					OutputFile: "/tmp/kfp_outputs/output_metadata.json",
+				},
+			},
+			true,
+			0,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fakeKubernetesClientset := &fake.Clientset{}
+			fakeMetadataClient := metadata.NewFakeClient()
+			bucket, err := blob.OpenBucket(nil, "gs://ml-pipeline-test")
+			assert.Nil(t, err)
+			bucketConfig, err := objectstore.ParseBucketConfig("gs://ml-pipeline-test/pipeline-root/")
+			assert.Nil(t, err)
+			addNumbersExecutorArgs := []string{"-c", "python3 -m venv .venv && source .venv/bin/activate\nexport CLUSTER_SPEC=\"{\\\"task\\\":{\\\"type\\\":\\\"workerpool0\\\",\\\"index\\\":0,\\\"trial\\\":\\\"TRIAL_ID\\\"}}\"\nif ! [ -x \"$(command -v pip)\" ]; then\n    python3 -m ensurepip || python3 -m ensurepip --user || apt-get install python3-pip\nfi\n\nPIP_DISABLE_PIP_VERSION_CHECK=1 python3 -m pip install --quiet     --no-warn-script-location 'kfp==2.0.0-rc.1' && \"$0\" \"$@\"\nprogram_path=$(mktemp -d)\nprintf \"%s\" \"$0\" > \"$program_path/ephemeral_component.py\"\npython3 -m kfp.components.executor_main                         --component_module_path                         \"$program_path/ephemeral_component.py\"                         \"$@\"\n", "\nimport kfp\nfrom kfp import dsl\nfrom kfp.dsl import *\nfrom typing import *\n\ndef add_numbers(a: int, b: int) -> int:\n    return a + b\n\n", "--executor_input", "{{$}}", "--function_to_execute", "add_numbers"}
```

Comment:
Given this args string, which contains only `{{$}}` but not something like ``{$.inputs.parameters['name']}`, the test doesn't actually cover the bug you're trying to fix, right?